File: Leetcode 3801-3900/3861/3861.java
// Leetcode 3861: Minimum Capacity Box
// https://leetcode.com/problems/minimum-capacity-box/
// Solved on 8th of March, 2026
class Solution {
    /**
     * Finds the index of the box with the minimum capacity that can fit the item.
     * @param capacity An array representing the capacities of available boxes.
     * @param itemSize The size of the item to be placed in a box.
     * @return The index of the box with the smallest capacity >= itemSize, or -1 if none fit.
     */
    public int minimumIndex(int[] capacity, int itemSize) {
        int bestIndex = -1;
        int minCapacity = Integer.MAX_VALUE;
        for (int i = 0; i < capacity.length; i++) {
            if (capacity[i] >= itemSize && capacity[i] < minCapacity) {
                minCapacity = capacity[i];
                bestIndex = i;
            }
        }
        return bestIndex;
    }
}

File: Leetcode 3801-3900/3861/3861.py
# Leetcode 3861: Minimum Capacity Box
# https://leetcode.com/problems/minimum-capacity-box/
# Solved on 8th of March, 2026
class Solution:
    def minimumIndex(self, capacity: list[int], itemSize: int) -> int:
        """
        Finds the index of the box with the smallest capacity that can fit the item.

        :param capacity: A list of integers representing the capacity of each box.
        :param itemSize: An integer representing the size of the item to be placed.
        :return: The index of the box with the minimum sufficient capacity, or -1 if no box fits.
        """
        bestIndex = -1
        minCapacity = float('inf')

        for i in range(len(capacity)):
            if capacity[i] >= itemSize and capacity[i] < minCapacity:
                minCapacity = capacity[i]
                bestIndex = i

        return bestIndex